### PR TITLE
Refactor classes for materialized view partititon filtering

### PR DESCRIPTION
**Pull request by Sullivan-Patrick**
Nonfunctional changes to make https://github.com/prestodb/presto/pull/18018 more readable

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java [hunk 1]
```diff
@@ -743,5 +737,25 @@ private RowExpression convertToRowExpression(Expression expression, Scope scope)
                     metadata.getFunctionAndTypeManager(),
                     session);
         }
+        private Scope extractScope(Table table, QuerySpecification node, Expression whereClause)
```
Comment (jainxrohit) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java:740]: leave a new line break

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java [hunk 2]
```diff
@@ -1309,6 +1301,46 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
...
+            // TODO: discover columns lazily based on where they are needed (to support connectors that can't enumerate all tables)
+            ImmutableList.Builder<Field> fields = ImmutableList.builder();
+            for (ColumnMetadata column : tableMetadata.getColumns()) {
+                fields.add(createField(table, column, name));
+            }
+
+            return createAndAssignScope(table, scope, fields.build());
+        }
+
+        private Field createField(Table table, ColumnMetadata columnMetadata, QualifiedObjectName name)
```
Comment (jainxrohit) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java:1331]: The method is very trivial, probably not required.

**Review by jainxrohit — COMMENTED** [review]
Can we make commit message more meaningful? Not really a fan of saying something for "upcoming commit"

**Review by jainxrohit — APPROVED** [review]

**Review by highker — COMMENTED** [review]
@pranjalssh, could you help to review this PR and request my review after approval?

**Review by pranjalssh — APPROVED** [review]

**Review by highker — APPROVED** [review]

**Pull request merged**